save-backend/src/main/kotlin/org/cqfn/save/backend/security/ProjectPermissionEvaluator.kt

```diff
@@ -36,6 +41,10 @@ class ProjectPermissionEvaluator {
 
     private fun Authentication.hasRole(role: Role): Boolean = authorities.any { it.authority == role.asSpringSecurityRole() }
 
-    private fun hasWriteAccess(userId: Long?, project: Project) =
-            userId != null && (project.userId == userId || userId in project.adminIdList())
+    private fun hasWriteAccess(userId: Long?, project: Project): Boolean {
+        val adminIdList = lnkUserProjectRepository.findByProject(project)
+            .filter { it?.role == Role.ADMIN }
+            .map { it?.user?.id }
```

Comment:
This logic can be encapsulated in some service (actually, it can be `ProjectService` itself). I'm also concerned, that this will mean a lot of DB queries, but optimizing this can be a separate issue. For now, you can at first check if the user is project owner, and only if not - query the repository

Reply:
Moved to lnkUserProjectService